_studio/shared/src/mfx_umc_alloc_wrapper.cpp

```diff
@@ -1049,12 +1049,12 @@ mfxStatus mfx_UMC_FrameAllocator::PrepareToOutput(mfxFrameSurface1 *surface_work
     surface.Info.Shift = m_IsUseExternalFrames ? m_extSurfaces[index].FrameSurface->Info.Shift : m_frameDataInternal.GetSurface(index).Info.Shift;
 
     //Performance issue. We need to unlock mutex to let decoding thread run async.
-    guard.Unlock();
+    guard.unlock();
```

Comment:
doesn't really makes much sense for me. If it is ```recursive_mutex``` it is still can be locked after this ```unlock``` call, so what is the point of unlocking it?

Reply:
I guess this unlock not related recursive mutex lock (DoFastCopyExtended() locks CommonCore::m_guard).
In my understanding unlock need for improve performance: another thread can run PrepareToOutput during execute CM copy

Reply:
I don't understand what you are talking about, you unlock here a ```recursive_mutex``` locked on line 958

Reply:
right, this unlock need for improve performance: another thread can run PrepareToOutput() during execute CM copy (DoFastCopyWrapper())

Reply:
Another thread can't run anything if this ```mutex``` is locked. Since you are using ```recursive_mutex``` after calling ```unlock``` is still can be locked and thought can't be used by another thread. That's why combination of ```recursive_mutex``` and manual ```unlock``` doesn't make sense for me